src/streams.c

```diff
@@ -125,23 +151,41 @@ Obj FuncREAD_ALL_COMMANDS( Obj self, Obj stream, Obj echo )
                 resultCapacity += 16;
                 GROW_PLIST(resultList, resultCapacity);
             }
-            result = NEW_PLIST( T_PLIST, 3 );
-            SET_LEN_PLIST(result, 1);
+            if (printToOutput == True) {
+                outstreamStringCopy = CopyToStringRep(outstreamString);
+                SET_LEN_STRING(outstreamString, 0);
+                result = NEW_PLIST(T_PLIST, 5);
+                SET_LEN_PLIST(result, 5);
+                SET_ELM_PLIST(result, 5, outstreamStringCopy);
+            }
+            else {
+                result = NEW_PLIST(T_PLIST, 3);
+                SET_LEN_PLIST(result, 3);
+            }
             SET_ELM_PLIST(result, 1, False);
             SET_ELM_PLIST(resultList, resultCount, result );
             SET_LEN_PLIST(resultList, resultCount );
```

Comment:
This code is missing a call to `CHANGED_BAG(resultList)` (this is an existing bug in it, not caused by this PR).

Even better would be to stop mucking around with low level calls like SET_ELM_PLIST and GROW_PLIST, and instead use PushPlist or AssPlist